allennlp/semparse/domain_languages/wikitables_language.py

```diff
@@ -0,0 +1,655 @@
+    def filter_date_greater(self, rows: List[Row], column: DateColumn, filter_value: Date) -> List[Row]:
+        cell_row_pairs = self._get_date_row_pairs_to_filter(rows, column.name)
+        return [row for cell_value, row in cell_row_pairs if cell_value > filter_value]
+
+    def filter_date_greater_equals(self, rows: List[Row], column: DateColumn, filter_value: Date) -> List[Row]:
+        cell_row_pairs = self._get_date_row_pairs_to_filter(rows, column.name)
+        return [row for cell_value, row in cell_row_pairs if cell_value >= filter_value]
+
+    def filter_date_lesser(self, rows: List[Row], column: DateColumn, filter_value: Date) -> List[Row]:
+        cell_row_pairs = self._get_date_row_pairs_to_filter(rows, column.name)
+        return [row for cell_value, row in cell_row_pairs if cell_value < filter_value]
+
+    def filter_date_lesser_equals(self, rows: List[Row], column: DateColumn, filter_value: Date) -> List[Row]:
+        cell_row_pairs = self._get_date_row_pairs_to_filter(rows, column.name)
+        return [row for cell_value, row in cell_row_pairs if cell_value <= filter_value]
+
+    def filter_date_equals(self, rows: List[Row], column: DateColumn, filter_value: Date) -> List[Row]:
+        cell_row_pairs = self._get_date_row_pairs_to_filter(rows, column.name)
+        return [row for cell_value, row in cell_row_pairs if cell_value == filter_value]
+
+    def filter_date_not_equals(self, rows: List[Row], column: DateColumn, filter_value: Date) -> List[Row]:
+        cell_row_pairs = self._get_date_row_pairs_to_filter(rows, column.name)
+        return [row for cell_value, row in cell_row_pairs if cell_value != filter_value]
+
+    # These two are similar to the filter methods above, but operate on strings obtained from the
+    # question, instead of dates or numbers.  So they check for whether the string value is present
+    # in the cell or not, instead of using a numerical / date comparator.
+    def filter_in(self, rows: List[Row], column: StringColumn, filter_value: str) -> List[Row]:
+        # Assuming filter value has underscores for spaces. The cell values also have underscores
+        # for spaces, so we do not need to replace them here.  Also, we need to remove the
+        # "string:" that was prepended to the entity name in the language.
+        filter_value = filter_value.lstrip('string:')
+        return [row for row in rows if filter_value in row.values[column.name]]
+
+    def filter_not_in(self, rows: List[Row], column: StringColumn, filter_value: str) -> List[Row]:
+        # Assuming filter value has underscores for spaces. The cell values also have underscores
+        # for spaces, so we do not need to replace them here.  Also, we need to remove the
+        # "string:" that was prepended to the entity name in the language.
+        filter_value = filter_value.lstrip('string:')
+        return [row for row in rows if filter_value not in row.values[column.name]]
+
+    @predicate
```

Comment:
May be move the generic predicates and column-type specific ones into two separate groups, and add a comment above the column-type specific group saying that you added those using `add_predicate` in the `__init__` method.